source/backtraceMetric.ts

```diff
@@ -0,0 +1,226 @@
+import { BacktraceClientOptions } from '.';
+import { uuid } from './utils';
+
+declare const __VERSION__: string;
+
+/**
+ * Function to create instance of BacktraceMetric and ping.
+ */
+export default function backtraceMetric(
+  configuration: BacktraceClientOptions,
+): void {
+  new BacktraceMetric(configuration).ping();
+}
+
+/**
+ * Handles Backtrace Metrics.
+ * TODO: Consider converting to a static class since an instance is only created to call a single function once.
+ */
+export class BacktraceMetric {
+  public universe;
+  public token;
+  public timeout = 15000; // fifteen seconds in miliseconds
+
+  /** Seconds since epoch. */
+  public readonly CURRENT_TIMESTAMP: number = Math.floor(
+    new Date().getTime() / 1000,
+  );
+  /** Thirty minutes in seconds. */
+  public readonly THIRTY_MINUTES = 1800;
+
+  public readonly userAgent = navigator.userAgent;
+  public readonly applicationName = 'backtrace-js';
+  public readonly applicationVersion = __VERSION__;
+
+  // These fields are stored in localStorage for persistence.
+  public sessionId = this.getSessionId();
+  public lastActive = this.getLastActive();
+  public sessionStart = this.getSessionStart();
+
+  private DIRECTORY = 'events-test'; // api endpoint ex. 'https://<directory>.backtrace.io/api/...'. Should just be 'events'
+
+  constructor(configuration: BacktraceClientOptions) {
+    const universe = ''; // get universe name from configuration if possible. If not, pass in / access universe name somewhere.
+
+    if (!configuration.endpoint) {
+      throw new Error(`Backtrace: missing 'endpoint' option.`);
+    }
+    if (!configuration.token) {
+      throw new Error(`Backtrace: missing 'token' option.`);
+    }
+
+    this.universe = universe;
+    this.token = configuration.token;
+    this.timeout = configuration.timeout;
+  }
+
+  /**
+   * Handle sessions and events. When pinged / called, will create or manage current session and send unique/summed events
+   * when appropriate.
+   */
+  public ping() {
```

Comment:
I think we should consider different strategy for the ping method. Let's use modern javascript libraries like react to describe the problem.

As you know we have a concept of router, store etc. In some of the websites we don't need to reload anything for a long time. 

Another example: Unity webgl game. 

Based on those two examples I see the reason why we should automatically send it after some time interval. Right now we send data only when someone refresh the website (so overall the backtrace-client integration is recreated). We should consider using something like timeInterval to send data automatically.

Reply:
Updated code, there is a heartbeat that runs every 1 minute.